### PR TITLE
Applied windows build fixes from 46df635

### PR DESCRIPTION
**Pull request by Meisaka**

### CODE REVIEW

File: include/systems/rewindable-map.hpp [hunk 1]
```diff
@@ -279,7 +279,7 @@ class RewindableMap final {
      */
     History<K,V> Pull(const Timepoint& frame_requested, Timepoint& last_received) const {
         auto el1 = backward_data.GetHistoryData(frame_requested, last_received);
-        return { std::move(el1), forward_data.PopSync(frame_requested, last_received) };
+        return History<K, V>(std::move(el1), forward_data.PopSync(frame_requested, last_received));
```
Comment (milesrout) [inline include/systems/rewindable-map.hpp:282]: If VS doesn't support basic C++11 features then it shouldn't be used.

Comment (Meisaka) [inline include/systems/rewindable-map.hpp:282]: This particular line would crash the MSVC CL compiler, so... yeah, that could be considered not supported.

Comment (adam4813) [inline include/systems/rewindable-map.hpp:282]: I am guessing this is due to MSVC not auto-generating move constructors.

**Comment by catageek** [comment]
Tested with gcc-4.8.2 and Clang 3.3

**Pull request merged**